src/cdk-experimental/popover-edit/lens-directives.ts

```diff
@@ -0,0 +1,144 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Directive, ElementRef, EventEmitter, Input, OnInit, Output} from '@angular/core';
+import {closest, Destroyable, EDIT_PANE_SELECTOR} from './common';
+import {EditRef} from './edit-ref';
+
+export type PopoverEditClickOutBehavior = 'close' | 'submit' | 'nothing';
+
+/**
+ * A directive that attaches to a form within the edit lens.
+ * It coordinates the form state with the table-wide edit system and handles
+ * closing the edit lens when the form is submitted or the user clicks
+ * out.
+ */
+@Directive({
+  selector: 'form[cdkEditControl]',
+  host: {
+    '(ngSubmit)': 'onSubmit()',
+    '(keydown.enter)': 'onEnter(true)',
+    '(keyup.enter)': 'onEnter(false)',
+    '(keyup.escape)': 'onEscape()',
+    '(document:click)': 'onDocumentClick($event)',
+  },
+  providers: [EditRef],
+})
+export class CdkEditControl extends Destroyable implements OnInit {
+  /**
+   * Specifies what should happen when the user clicks outside of the edit lens.
+   * The default behavior is to close the lens without submitting the form.
+   */
+  @Input() cdkEditControlClickOutBehavior = 'close';
```

Comment:
What are the types that this can have?

Also, you can rename the variable so it doesn't have `cdkEditControl...` in the name, but provide the input name in the `@Input(cdkEditControl...)`

Reply:
Defined the type for it and forgot to use it - fixed